ivy/functional/frontends/builtins/func_wrapper.py

```diff
@@ -0,0 +1,88 @@
+import functools
+from typing import Callable, Iterable
+from importlib import import_module
+
+import ivy
+
+
+def _to_ivy_array(x):
+    # if x is any frontend array including frontend list
+    # then extract the wrapped ivy array
+    if hasattr(x, "ivy_array"):
+        return x.ivy_array
+
+    # convert native arrays and lists to ivy arrays
+    elif isinstance(x, ivy.NativeArray):
+        return ivy.array(x)
+
+    return x
+
+
+def _infer_return_array(x: Iterable) -> Callable:
+    # get module path
```

Comment:
Could you add a docstring that explains what this decorator does overall? It is not very obvious when it should be used.

Reply:
This function should always be used within `@to_ivy_arrays_and_back` decorator to infer the return array type. Let's consider the `jax` frontend. In this case, if decorator receives `jax.Array`, its frontend implementation or `list` the output will always be converted to a frontend implementation of `jax.Array`. However, in case of `builtins` the behaviour changes to: `almost_any_array -> ivy.Array -> almost_any_array`. In other words, we need to track what array to convert back to. So, if our built-in function recieves `np.ndarray` or its frontend implementation, the output will be converted to `np_frontend.ndarray` and so on. It also has a default behaviour. By default we convert the output to a frontend implementation of array which corresponds to current backend. This was done to cover the cases, when function accepts scalar arguments, but returns an array. For example, `range`. But yeah I will add a docstring.